### DISCUSSION TITLE
http/latency

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#208] [General]
# http/latency

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/http/latency.html

**Comment by JiHyoz** [comment]
如图 2-1 请求阶段分析所示，这些阶段共需要 4 个 RTT（Round-Trip Time，往返时间）[1] = 1 RTT（DNS Lookup，域名解析）+ 1 RTT（TCP Handshark，TCP 握手）+ 2 RTT（SSL Handshark，SSL 握手）+ 1 RTT（Data Transfer，HTTP 内容请求传输）。

**Reply by JiHyoz** [reply]
这是五个RRT吧

**Reply by isno** [reply]
谢谢提醒。

**Comment by therainisme** [comment]
表 2-3 中，关于 TTFB 的描述“服务器从接收请求到开始到收到第一个字节的耗时”，读起来感觉有歧义，缺少了主语。读起来像“......服务器收到第一个字节的耗时”。

而实际上是“......到客户端收到第一个字节的耗时”。

**Reply by isno** [reply]
谢谢，已修改

**Comment by AnotherJack** [comment]
单词拼写错误，Handshark -> Handshake

**Reply by isno** [reply]
谢谢，已修改。

**Comment by lmq3030** [comment]
> 服务器响应第一个字节到全部传输完成耗时
如果 response payload 很大，需要多个 packet 才能传输完成，那么是不是就无法用来计算“服务器处理请求耗时”

**Reply by lmq3030** [reply]
需要多个 packet 才能传输完成 => 需要 client side ack packet delivery

**Reply by isno** [reply]
依旧可以计算“服务器处理请求耗时”。

**Comment by DengLZ** [comment]
"TTFB = time_starttransfer - time_appconnect	  服务器从接收请求到开始到收到第一个字节的耗时".    表格中的这个计算方式不对吧. 我看TTFB是这么定义的. https://developer.mozilla.org/zh-CN/docs/Glossary/Time_to_first_byte  . 按照这个定义计算 TTFB = time_starttransfer.

**Reply by isno** [reply]
这里是我弄错了，已修改。感谢！
